### PR TITLE
Updated README and manual to include new features

### LINKED ISSUE #64: FSM: Write out inputs/outputs/internals
They will need passing through the FSM translation and writing to the `.sg` file.

### PR DESCRIPTION
**Pull request by jrbeaumont**
This should allow us to close #64 #60 and #36.

### CODE REVIEW

File: doc/manual.md [hunk 1]
```diff
@@ -17,49 +17,49 @@ information on the theory of concepts can be found
...
+Plato, as well as Workcraft are available
 for *Windows*, *Linux* and *macOS*. The installation instructions for
 all of these operating systems are the same. We will be referring to
 directories using the
 forward-slash character (’/’) as a separator, however for *Windows*,
 replace this with a back-slash character (’\\’).
 
-If choosing to use the concepts tool on its own, this must be downloaded
-from the [GitHub repo](https://github.com/tuura/concepts). If you choose
+If choosing to use plato on its own, this must be downloaded
```
Comment (snowleopard) [inline doc/manual.md]: Let's keep Plato capitalised throughout the text (not in links though).

File: doc/manual.md [hunk 2]
```diff
@@ -109,6 +108,126 @@ install the concepts tool, simply run:
...
+	module Concept where
+
+	import Tuura.Concept.STG
+
+	circuit a b c = interface <> outputRise <> inputFall <> outputFall <> inputRise <> initialState
+	  where
+    	interface = inputs [a, b] <> outputs [c]
+
+	    outputRise = rise a ~> rise c <> rise b ~> rise c
+
```
Comment (snowleopard) [inline doc/manual.md]: The indentation seems all over the place, could you fix this?

File: doc/manual.md [hunk 3]
```diff
@@ -109,6 +108,126 @@ install the concepts tool, simply run:
...
+Line 2
+
+This must also remain in all concept files, before any concepts begin to
+be defined. Importing this module means that the standard operators
+and existing gates/protocols can be used.
+
+Line 3
+
+This is where a user can begin to define their concepts. “`circuit`” must
```
Comment (snowleopard) [inline doc/manual.md]: I think backticks are enough to signify a code snippet: `circuit`. No need for adding quotes around that.

File: doc/manual.md [hunk 4]
```diff
@@ -204,20 +336,164 @@ standard type used as input to tools, such as Petrify, Mpsat, and Workcraft. The
...
+.end
+```
+
+This produces an FSM in the *.sg* format. Similar to the STG *.g* format, this
+can be copy-and-pasted into a file and saved with this *.sg* and used as an 
+input to various tools, including Workcraft.
+
+Again, any errors that occur during the translation process will produce 
+errors referring to the problematic lines of signals of the concepts that are 
+problematic.
```
Comment (snowleopard) [inline doc/manual.md]: Looks like there is some text duplication here.

File: doc/manual.md [hunk 5]
```diff
@@ -562,11 +727,21 @@ the same as in the standard `handshake` concepts, but this includes
 initial states for the signals. For this concept, both signals will
 be initialised to 1, or high.
 
+`never [(Transition) a, (Transition) b, ...]` 
+
+A concept used to define lists of signal transitions which cannot all have 
+occured at the same time. `(Transition)` in this 
+case will be `rise` or `fall`. These lists are used to descripe states of the
+system which do not hold for the invariant. For example, the concept
```
Comment (snowleopard) [inline doc/manual.md]: I can't parse the sentence: "These lists are used to descripe states of the system which do not hold for the invariant."

(And the typo in `descripe` is not the reason.)

These lists are used to describe the invariant of the system, by specifying classes of states that must be unreachable. -- perhaps, something like this?

**Comment by snowleopard** [comment]
@jrbeaumont See some comments above.

**Comment by jrbeaumont** [comment]
@snowleopard: I have fixed those issues. Let me know if you find any further ones.

**Comment by snowleopard** [comment]
Thanks, merged now!

**Comment by jrbeaumont** [comment]
Thank you.

**Pull request merged**